src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -0,0 +1,133 @@
+package org.hammerlab.guacamole.commands
+
+import org.apache.spark.SparkContext
+import org.apache.spark.util.StatCounter
+import org.hammerlab.guacamole.{ DistributedUtil, SparkCommand, Common }
+import org.hammerlab.guacamole.reads.{ PairedRead, Read, MappedRead }
+import org.kohsuke.args4j.{ Option => Args4JOption }
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Structural Variant caller
+ *
+ * This currently looks for regions with abnormally large insert sizes. This typically indicates
+ * that there's been a large deletion.
+ *
+ * It does this by finding the average insert size of all mate pairs which overlap each 25-base pair "block" of the
+ * genome. If this average is significantly different than the average insert size genome-wide, then it's evidence that
+ * there may be a structural variant.
+ * 
+ * The output is a text file of genomic ranges where there's some evidence of an SV.
+ */
+object StructuralVariant {
+
+  val MAX_INSERT_SIZE = 25000
+  val BLOCK_SIZE = 25
+
+  protected class Arguments extends Common.Arguments.Reads with DistributedUtil.Arguments {
+    @Args4JOption(name = "--filter-contig", usage = "Filter to alignments where either mate is in this contig.")
+    var filterContig: String = ""
+
+    @Args4JOption(name = "--output", usage = "Path to output text file.")
+    var output: String = ""
+  }
+
+  case class Location(
+      contig: String,
+      position: Long) {
+  }
+
+  case class GenomeRange(
```

Comment:
Just noting that equivalents of these two case classes seem to proliferate around genomics codebases.

Using BDG [`Feature`](https://github.com/bigdatagenomics/bdg-formats/blob/de290eb5e5e237e13ee20ca9b162744888721041/src/main/resources/avro/bdg.avdl#L710)s / [`ReferenceRegion`](https://github.com/bigdatagenomics/adam/blob/a68a7e1ad3d970111d007b6a4011365e15e213a6/adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala#L125)s would give you writing-to-Parquet for free, instead of a text file, fwiw.